### ISSUE TITLE
Feature: Support reading snappy compressed parquet file

### ISSUE THREAD

**Issue by kesavkolla** [datafuselabs/databend#9417]
**read_parquet table function should support different compression options**

Usually parquet files are compressed using snappy or zstd. The read_parquet table function should support compression options to deal with popular compression algorithms.

**Comment by BohuTANG** [comment]
For the `read_parquet` function, it already supports `snappy|zstd` compression:
https://github.com/datafuselabs/databend/blob/9edfb5b1e25b4e172dd28d688a60763b1c7d4120/src/query/storages/parquet/src/parquet_part.rs#L80-L93

The parquet reader will automatically detect underlying compression what's it using:
https://github.com/datafuselabs/databend/blob/9edfb5b1e25b4e172dd28d688a60763b1c7d4120/src/query/storages/parquet/src/table_function/read.rs#L178-L183

So if your parquet file with snappy or zstd, it should work well.

**Comment by kesavkolla** [comment]
Hmm... Then there is a bug in the system. Here is what I have done:

Ran a databend container by using:

```shell
docker run -d --name databend --net=host datafuselabs/databend
```

Once it started I modifed the databend-query.toml with allow_insecure = true
I copied a snappy compressed parquet file into /tmp inside container.

I connected using mysql CLI and confirmed setting is in effect:

MySQL [(none)]> select * from system.configs where name='allow_insecure';
+---------+----------------+-------+-------------+
| group   | name           | value | description |
+---------+----------------+-------+-------------+
| storage | allow_insecure | true  |             |
+---------+----------------+-------+-------------+
1 row in set (0.010 sec)

When I ran the following query:

```sql
select * from read_parquet('/tmp/part-00000-46637011-9426-41d3-9988-c00d56007803-c000.snappy.parquet') limit 10;
```
It just hung nothing is returned. I waited almost 4 min then hand to kill the mysql cli to come out.

**Comment by kesavkolla** [comment]
This is a valid parquet file. I have used datafusion CLI to run some query operations on this file.

**Comment by sundy-li** [comment]
Ok, we will test this.  Can you share the example parquet file?

**Comment by kesavkolla** [comment]
Here is an example parquet file I was trying. You have to unzip the file first.
[part-00000-f6337bce-7fcd-4021-9f9d-040413ea83f8-c000.snappy.parquet.zip](https://github.com/datafuselabs/databend/files/10326739/part-00000-f6337bce-7fcd-4021-9f9d-040413ea83f8-c000.snappy.parquet.zip)

**Comment by sundy-li** [comment]
Thanks, I've reproduced it, it's a bug!

**Comment by kesavkolla** [comment]
Is it fixed now?

**Comment by sundy-li** [comment]
> Is it fixed now?

Not yet, it may take some time to support Struct datatype in `read_parquet` function.  Happy new year!

**Comment by RinChanNOWWW** [comment]
#9486 and #9537 may solved this. PTAL @kesavkolla

**Comment by kesavkolla** [comment]
@RinChanNOWWW Still error persists. Now the error is different:

```
ERROR 1105 (HY000): Code: 1104, displayText = called `Result::unwrap()` on an `Err` value: OutOfSpec("The children must have an equal number of values.\n                         However, the values at index 4 have a length of 75479, which is different from values at index 0, 20000.").
```

**Comment by RinChanNOWWW** [comment]
> @RinChanNOWWW Still error persists. Now the error is different:
> 
> ```
> ERROR 1105 (HY000): Code: 1104, displayText = called `Result::unwrap()` on an `Err` value: OutOfSpec("The children must have an equal number of values.\n                         However, the values at index 4 have a length of 75479, which is different from values at index 0, 20000.").
> ```

What SQL did you use?

**Comment by RinChanNOWWW** [comment]
I found the bug, and it will be fixed in #9563
